docker/Dockerfile

```diff
@@ -1,41 +1,29 @@
-FROM registry.service.dsd.io/ruby:2.6.3
+FROM ruby:2.6.3-alpine3.8
 
-# add official nodejs repo
-RUN curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - && \
-        echo 'deb https://deb.nodesource.com/node jessie main' > /etc/apt/sources.list.d/nodesource.list
+# Install pre-requisites for building nokogiri & pg gems, as well
+# as the packages for converting and generating PDF's
+RUN apk --update add --virtual build_deps \
+    build-base ruby-dev libc-dev linux-headers \
+    libxml2-dev libxslt-dev \
+    postgresql-dev pdftk runit nodejs libreoffice redis git
 
-# install runit, nodejs and libreoffice (used by ADP to generate PDFs)
-# pdftk is used to fill in PDF forms for disc evidence download
-RUN apt-get update && apt-get install -y pdftk runit nodejs libreoffice redis-server && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/* && rm -fr *Release* *Sources* *Packages* && \
-    truncate -s 0 /var/log/*log
+RUN addgroup -g 1000 -S appgroup && \
+    adduser -u 1000 -S appuser -G appgroup
 
-# SSH proxy settings
-ENV SSH_AUTH_SOCK /tmp/ssh-auth
-ENV SSH_AUTH_PROXY_PORT 1234
+WORKDIR /app
+COPY . .
+RUN mkdir -p /app/tmp
 
-RUN mkdir -p /usr/src/app
 RUN bundle config --global without test:development
-
-WORKDIR /usr/src/app
-
-COPY Gemfile /usr/src/app/
-COPY Gemfile.lock /usr/src/app/
-
-# Hack to install private gems
-#RUN socat UNIX-LISTEN:$SSH_AUTH_SOCK,fork TCP4:$(ip route|awk '/default/ {print $3}'):$SSH_AUTH_PROXY_PORT & bundle install
+RUN gem install bundler -v 1.17.3
```

Comment:
to speed up builds it is, apparently, a good idea to only copy dependencies over - found this [9-common-dockerfile-mistakes](https://runnable.com/blog/9-common-dockerfile-mistakes)

Also we should probably be bundle installing as the user (i hit problems when i didn't). In total here is the way i did it in the dummy app

```
######################
# DEPENDENCIES START #
######################
# NOTE: all COPY commands set file permissions to root
COPY Gemfile Gemfile
COPY Gemfile.lock Gemfile.lock

# change to the added user
USER appuser
RUN gem install bundler -v 1.17.3

# remove test development gems from default bundle install
RUN bundle config --global without test:development

# nokogiri install problems
# https://github.com/gliderlabs/docker-alpine/issues/53
#
# Use libxml2-dev, libxslt-dev packages, installed above, for building nokogiri
RUN bundle config build.nokogiri --use-system-libraries

RUN bundle install

####################
# DEPENDENCIES END #
####################
```